test/assetGiveaway/assetGiveaway.test.ts

```diff
@@ -0,0 +1,319 @@
+import {ethers} from 'hardhat';
```

Comment:
We should have test when the assetHolder is another address.
this is likely to be how we setup our claimtokens as this will be cheaper gas wise.

Actually thinking about it, with _assetHolder == address(0), we need to get to know the asset id ahead of time.
it might be a good idea to let the admin set the metkle tree root hash after deployment, maybe only once?

Reply:
Sounds good